### DISCUSSION TITLE
Depth of GLTextItem

### DISCUSSION THREAD

**Discussion by sheodun** [pyqtgraph/pyqtgraph#2075] [Q&A] [answered]
Hi everyone!

I have a quick question about the depth of the gl.GLTextItem.

It appears that the TextItem will always remain at a given depth with respect to GLMeshItems as given by the depth value. In other words, depending on the order that the text or mesh item is added to the glViewWidget (or the depth values chosen), the text will **always** be drawn in that order.

Is there a way to set the depth value of the text with respect to the meshitems on the basis of which is closer automatically?

**Examples:**
_Side on view (working as intended):_
![image](https://user-images.githubusercontent.com/90875475/141513608-14aa3309-c3fb-4ebd-9ecf-8792928b57a1.png)
_Top-down view (working as intended):_
![image](https://user-images.githubusercontent.com/90875475/141513631-8cb0cbc7-6576-47eb-9c67-472aecb9ea6b.png)
_Bottom-up view (working not as intended, the text should be behind the bigger atom):_
![image](https://user-images.githubusercontent.com/90875475/141513661-17bc5a9e-5f6c-4332-a0c3-56b7c6bc0145.png)

Many thanks!

**Comment by sheodun** [comment, marked as answer]
I managed to solve this so thought I'd leave a message here for anyone else who needs to know the answer.

I subclassed the GLViewWidget's PaintGL method to cycle through the plotted items, get their x, y and z coordinates (which you can get from the 4x4 QMatrix, elements {[0,3], [1,3] and [2,3]}) and compute their distances from the camera. Order the distances and then set the item's depthValue to an increasing value for the closer items. Then call the original PaintGL method!

**Reply by j9ac9k** [reply]
Hi @sheodun I'm glad you sorted it out; I meant to reply earlier, but truth be told I didn't have a good answer off the top of my head.

Appreciate you describing your solution!
